controllers/clustergroupupgrade_controller.go

```diff
@@ -315,86 +313,82 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
-					var reconcileSooner bool
-					// Add the needed cluster names to upgrade to the appropriate placement rule.
-					reconcileSooner, err = r.remediateCurrentBatch(ctx, clusterGroupUpgrade)
+				if clusterGroupUpgrade.Status.Status.CurrentBatch < len(clusterGroupUpgrade.Status.RemediationPlan) {
+					// Check if current policies have become compliant and if new policies have to be applied.
+					var isBatchComplete bool
+					isBatchComplete, err = r.getNextRemediationPoliciesForBatch(ctx, clusterGroupUpgrade)
 					if err != nil {
 						return
 					}
-					if reconcileSooner {
-						nextReconcile = requeueWithShortInterval()
-					}
 
-					// Check if this batch has timed out
-					if !clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.IsZero() {
-
-						currentBatchTimeout := utils.CalculateBatchTimeout(
-							clusterGroupUpgrade.Spec.RemediationStrategy.Timeout,
-							len(clusterGroupUpgrade.Status.RemediationPlan),
-							clusterGroupUpgrade.Status.Status.CurrentBatch,
-							clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.Time,
-							clusterGroupUpgrade.Status.Status.StartedAt.Time)
-
-						if time.Since(clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.Time) > currentBatchTimeout {
-							// Check if this was a canary or not
-							if len(clusterGroupUpgrade.Spec.RemediationStrategy.Canaries) != 0 &&
-								clusterGroupUpgrade.Status.Status.CurrentBatch <= len(clusterGroupUpgrade.Spec.RemediationStrategy.Canaries) {
-								r.Log.Info("Canaries batch timed out")
-								meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
-									Type:    "Ready",
-									Status:  metav1.ConditionFalse,
-									Reason:  "UpgradeTimedOut",
-									Message: "The ClusterGroupUpgrade CR policies are taking too long to complete",
-								})
-							} else {
-								r.Log.Info("Batch upgrade timed out")
-								clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt = metav1.Time{}
-								if clusterGroupUpgrade.Status.Status.CurrentBatch < len(clusterGroupUpgrade.Status.RemediationPlan) {
-									clusterGroupUpgrade.Status.Status.CurrentBatch++
+					if isBatchComplete {
+						// If the upgrade is completed for the current batch, cleanup and move to the next.
+						r.Log.Info("[Reconcile] Upgrade completed for batch", "batchIndex", clusterGroupUpgrade.Status.Status.CurrentBatch)
+						r.cleanupPlacementRules(ctx, clusterGroupUpgrade)
+						clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt = metav1.Time{}
+
+						clusterGroupUpgrade.Status.Status.CurrentBatch++
+						nextReconcile = requeueImmediately()
+					} else {
+						// Add the needed cluster names to upgrade to the appropriate placement rule.
+						nextReconcile, err = r.remediateCurrentBatch(ctx, clusterGroupUpgrade)
+						if err != nil {
+							return
+						}
+
+						// Check if this batch has timed out
+						if !clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.IsZero() {
+							
```

Comment:
This line has trailing whitespace (tabs to indent, but it's a blank line so unneeded). The "go fmt" step that's done by ci-job then updates the file to remove this whitespace, which gets caught by the bundle-check (which looks for a "dirty" working tree after running the ci-job build steps)